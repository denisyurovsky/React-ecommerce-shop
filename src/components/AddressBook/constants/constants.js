export const LISTBOX_PADDING = 8;
export const DOUBLE_LISTBOX_PADDING = LISTBOX_PADDING * 2;
export const LIMIT_NUMBER_ITEMS = 8;
export const ITEM_SIZE_LG = 48;
export const ITEM_SIZE_MD = 36;
