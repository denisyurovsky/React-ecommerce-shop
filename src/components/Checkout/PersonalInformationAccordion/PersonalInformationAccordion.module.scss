@import "../../../assets/styles/colors.scss";

.accordion {
  width: 100%;
  border-top-left-radius: 15px !important;
  border-top-right-radius: 15px !important;
  margin-bottom: 15px;

  &::before {
    display: none;
  }
}

.accordionSummary {
  background-color: $greyMain !important;
}

.header {
  display: flex;
  align-items: center;
}

.icon {
  width: 30px !important;
  height: 30px !important;
  margin-right: 15px !important;
  transition: 1s;
}

.subtitle {
  font-size: 18px;
}

.body {
  display: flex;
  flex-direction: column;
}

.button {
  margin-top: 20px !important;
}
