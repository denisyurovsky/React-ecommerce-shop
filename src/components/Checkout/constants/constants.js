export const PANEL = {
  DELIVERY_ADDRESS: 'Delivery Address',
  PERSONAL_INFORMATION: 'Personal Information',
  PERSONAL_ADDRESS: 'Personal Address',
  PAYMENT_METHOD: 'Payment method',
  ORDER_CONFIRMATION: 'Order confirmation',
};
