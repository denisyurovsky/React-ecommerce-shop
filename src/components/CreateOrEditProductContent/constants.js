export const PRICE_ERROR = 'Only numbers';
export const DESCRIPTION_ERROR = 'Only latin characters';
export const LENGTH_ERROR = 'Should contain at least one symbol';
export const DISCOUNT_ERROR = 'Discount should be less than price';
export const SELECTION_ERROR = 'Choose category';
