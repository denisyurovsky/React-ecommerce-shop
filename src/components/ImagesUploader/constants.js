export const LOSS_OF_IMAGES = 'We lost images';

export const STATUS = {
  CROPPED: 'cropped',
  CROPPING: 'cropping',
  IDLE: 'idle',
};

export const TEST_IMG_URL = 'http://someURL';
