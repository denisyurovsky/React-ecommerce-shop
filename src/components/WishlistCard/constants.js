export const DEFAULT_LABEL_INPUT = 'Wishlist Name';
export const ERROR_LABEL_INPUT = 'Wishlist name should be unique';
