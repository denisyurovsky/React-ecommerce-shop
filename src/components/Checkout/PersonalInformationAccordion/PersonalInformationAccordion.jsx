import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
import ContactPageOutlinedIcon from '@mui/icons-material/ContactPageOutlined';
import EditIcon from '@mui/icons-material/Edit';
import {
  Accordion,
  AccordionDetails,
  Box,
  Typography,
  Button,
} from '@mui/material';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import PersonalInformation from '../../AddressBook/PersonalInformation/PersonalInformation';
import { PANEL } from '../constants/constants';
import AccordionSummary from '../helpers/AccordionSummary';

import styles from './PersonalInformationAccordion.module.scss';

const PersonalInformationAccordion = ({
  expanded,
  handleChangeAccordion,
  address,
  isPersonalInformationValid,
  handlePersonalInformationButton,
  setAddress,
  handleChange,
}) => {
  const { title, name, surname, phone } = address;
  const isPersonalInformationExpended = expanded === PANEL.PERSONAL_INFORMATION;

  return (
    <Accordion
      className={classNames({
        [styles.accordionSummary]: !isPersonalInformationExpended,
        [styles.accordion]: true,
      })}
      expanded={isPersonalInformationExpended}
      onChange={handleChangeAccordion(PANEL.PERSONAL_INFORMATION)}
    >
      <AccordionSummary
        expandIcon={
          isPersonalInformationExpended ? <ArrowDropDownIcon /> : <EditIcon />
        }
        data-testid="personalInformationSummery"
      >
        <Box className={styles.header}>
          <ContactPageOutlinedIcon className={styles.icon} />
          <Box>
            <Typography className={styles.subtitle} variant="subtitle2">
              Personal information
            </Typography>
            {expanded !== PANEL.PERSONAL_INFORMATION &&
              isPersonalInformationValid && (
                <Box className={styles.body}>
                  <Typography variant="caption" data-testid="fullName">
                    {title} {name} {surname}
                  </Typography>
                  <Typography variant="caption">{phone}</Typography>
                </Box>
              )}
          </Box>
        </Box>
      </AccordionSummary>
      <AccordionDetails>
        <PersonalInformation
          handleChange={handleChange}
          address={address}
          setAddress={setAddress}
        />
        <Button
          onClick={handlePersonalInformationButton}
          fullWidth
          className={styles.button}
          variant="contained"
          disabled={!isPersonalInformationValid}
          data-testid="personalInformationButton"
        >
          Proceed
        </Button>
      </AccordionDetails>
    </Accordion>
  );
};

PersonalInformationAccordion.propTypes = {
  handleChangeAccordion: PropTypes.func.isRequired,
  address: PropTypes.object.isRequired,
  handlePersonalInformationButton: PropTypes.func.isRequired,
  setAddress: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  expanded: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]).isRequired,
  isPersonalInformationValid: PropTypes.bool,
};

export default PersonalInformationAccordion;
