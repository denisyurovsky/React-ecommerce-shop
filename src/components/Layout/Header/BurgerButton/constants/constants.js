import { pathNames } from '../../../../../constants/pathNames';

export const LINKS = [
  {
    name: 'HOME',
    link: '/',
  },
  {
    name: 'PRODUCTS',
    link: pathNames.PRODUCTS,
  },
];
