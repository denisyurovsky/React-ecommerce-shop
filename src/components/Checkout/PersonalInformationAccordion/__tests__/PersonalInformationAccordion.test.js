import React from 'react';

import { EMPTY_ADDRESS } from '../../../../pages/AddressBookPage/constants/constants';
import renderWith from '../../../../test-utils/renderWith';
import { PANEL } from '../../constants/constants';
import PersonalInformationAccordion from '../PersonalInformationAccordion';

describe('PersonalInformationAccordion component', () => {
  it('should render valid snapshot', () => {
    const { asFragment } = renderWith(
      <PersonalInformationAccordion
        handleChangeAccordion={() => {}}
        handlePersonalInformationButton={() => {}}
        handleChange={() => {}}
        setAddress={() => {}}
        expanded={PANEL.PERSONAL_INFORMATION}
        isPersonalInformationValid={false}
        address={EMPTY_ADDRESS}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
