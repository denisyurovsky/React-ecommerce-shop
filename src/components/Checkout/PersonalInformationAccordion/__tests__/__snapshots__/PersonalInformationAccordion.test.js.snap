// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInformationAccordion component should render valid snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded MuiAccordion-gutters accordion css-1elwnq4-MuiPaper-root-MuiAccordion-root"
  >
    <div
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-24vt7b-MuiButtonBase-root-MuiAccordionSummary-root"
      data-testid="personalInformationSummery"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
      >
        <div
          class="header MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
            data-testid="ContactPageOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13.17 4 18 8.83V20H6V4h7.17M14 2H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V8l-6-6zm-2 12c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85-.85-.37-1.79-.58-2.78-.58-.99 0-1.93.21-2.78.58C8.48 15.9 8 16.62 8 17.43V18h8v-.57z"
            />
          </svg>
          <div
            class="MuiBox-root css-0"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 subtitle css-ati914-MuiTypography-root"
            >
              Personal information
            </h6>
          </div>
        </div>
      </div>
      <div
        class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            class="MuiAccordion-region"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
            >
              <div
                class="wrapperFullName MuiBox-root css-0"
              >
                <div
                  class="MuiFormControl-root title css-1nrlq1o-MuiFormControl-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    id="title"
                  >
                    Title
                  </label>
                  <div
                    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
                    data-testid="select-title"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby="title mui-component-select-title"
                      class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
                      id="mui-component-select-title"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </div>
                    <input
                      aria-hidden="true"
                      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      name="title"
                      tabindex="-1"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  class="wrapper MuiBox-root css-0"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root name css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="name"
                      id="name-label"
                    >
                      Name
                    </label>
                    <div
                      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
                    >
                      <input
                        aria-describedby="name-helper-text"
                        aria-invalid="false"
                        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                        id="name"
                        type="text"
                        value=""
                      />
                    </div>
                    <p
                      class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
                      id="name-helper-text"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </p>
                  </div>
                  <div
                    class="MuiFormControl-root MuiTextField-root surname css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="surname"
                      id="surname-label"
                    >
                      Surname
                    </label>
                    <div
                      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
                    >
                      <input
                        aria-describedby="surname-helper-text"
                        aria-invalid="false"
                        class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                        id="surname"
                        type="text"
                        value=""
                      />
                    </div>
                    <p
                      class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
                      id="surname-helper-text"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="mui-1"
                  id="mui-1-label"
                >
                  Phone
                </label>
                <div
                  class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-describedby="mui-1-helper-text"
                    aria-invalid="false"
                    class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    id="mui-1"
                    type="text"
                    value=""
                  />
                </div>
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-sizeMedium css-1d1r5q-MuiFormHelperText-root"
                  id="mui-1-helper-text"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </p>
              </div>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root Mui-disabled button css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
                data-testid="personalInformationButton"
                disabled=""
                tabindex="-1"
                type="button"
              >
                Proceed
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
